keps/sig-node/4753-expose-ownerref-downard-api/README.md

```diff
@@ -0,0 +1,659 @@
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Today when a pod wants to pass its `ownerReferences` to a new object it manage (a ConfigMap, a Secret, etc), it needs to call the API server to GET it's own ownerReferences and then pass it to the new object. 
+The pod then needs GET access on all the intermediate resources which it may not have, and giving it that access is a security risk as it can access other resources it should not have access to.
+This KEP proposes to expose the ownerReferences via `valueFrom` and downward API, so that the pod can directly pass its `ownerReferences` to the managed/created new object without having to call the API server.
+
+This KEP is intentionally scoped small, and intends to keep the scope just to `ownerReferences` and not to other fields of the owner object to prevent security risks.
+
+## Motivation
+
+DaemonSets can dynamically orphan and adopt pods. Consider a scenario where, on day 1, a pod named foobar-deadbeef93-xyz76 on node XYZ creates an object (e.g a ConfigMap) associated with node XYZ, then  it would be incorrect to set the object owner to the pod foobar-deadbeef93-xyz76 instead, it should be set to damonset foobar, to prevent the object from being garbage collected if the pod cease to exist. This issue arises when pods create resources like ConfigMaps or CustomResources and these resources are not correctly reassigned to the new pod, leading to orphaned objects. Ensuring that daemonset pods can pass their ownerReferences to new objects they manage would prevent this issue, as the ownership hierarchy would be preserved even when pods are recreated.
+
+For example, a CustomResource created by a pod managed by a DaemonSet may be unexpectedly garbage collected if the pod is deleted. This can disrupt system behavior, as the CustomResource is garbageCollected. By allowing the pod to inherit and pass down ownerReferences, the CustomResource would remain correctly managed by the DaemonSet(the owner), avoiding disruptions.
+
+### Goals
```

Comment:
do we only need one level of ownerReference? Would it ever be needed to know the owner of the owner of the owner? I cannot tell from the motivation section whether one level will always be enough

Reply:
Well, I don't have a strong use case for that. If you have a use case maybe we can discuss it and integrate it.

Reply:
one option is to just understand how it can be extended to the owner of an owner in future. To make sure we are not making it unnatural to be extended this way in future